### ISSUE TITLE
gcsfuse fs hangs when several servers write to the same bucket

### ISSUE THREAD

**Issue by xmedeko** [GoogleCloudPlatform/gcsfuse#193]
Hi,
we process videos by ffmpeg and write these videos (1-2GB) to the fs mounted by gcsfuse (19.0, default options). Usually, we start just a few servers (max 10) , but recently, I have tried to start about 20 servers writing to the same bucket (but different "directories") and the gcsfuse volume has hanged. Even `ls /mnt/bucket` has not finished. I have tried `ls` from a few servers (even a server not writing anything to the bucket) and all has hanged. I have stopped most of the servers, and the `ls` has started to work again. I do not know if the problem was just in one particular server, or the problem was that so many servers were writing at one time.

Update: may this be caused by [Google API flood protection](https://developers.google.com/drive/v3/web/handle-errors#500_backend_error)?

**Comment by jacobsa** [comment]
* Are you sure this is not just saturation of network bandwidth? You are likely
  sending lots of data to GCS.

* It would be good to see some debug logging here. Try `--foreground --debug_fuse`.

**Comment by xmedeko** [comment]
I've forgotten important detail: when the gcsfuse has hanged, the `gsutil ls` was working and quick on the same machine.

I'll try to mount one machine with `--foreground --debug_fuse` when I experience similar problems.

**Comment by xmedeko** [comment]
Hmm, strange, I've tried yesterday to start 15 servers and have not experience any problems with `ls /mnt/bucket`. Maybe the problems was in the Google Cloud backend and not gcsfuse?

I have tried `gcsfuse --debug_http` and then `ls` around, and see it uses both [list](https://cloud.google.com/storage/docs/json_api/v1/objects/list) and [get](https://cloud.google.com/storage/docs/json_api/v1/objects/get) operation. maybe the backend has been stuck in the `get` operation, while the `list` has been working? I think this issue cannot be solved without knowledge what's going on in the Storage backend under heavy write loads.
